pkg/mount/mounter_linux.go

```diff
@@ -4,15 +4,48 @@ import (
 	"syscall"
 )
 
-func mount(device, target, mType string, flag uintptr, data string) error {
-	if err := syscall.Mount(device, target, mType, flag, data); err != nil {
-		return err
+const (
+	// ptypes is the set propagation types.
+	ptypes = syscall.MS_SHARED | syscall.MS_PRIVATE | syscall.MS_SLAVE | syscall.MS_UNBINDABLE
+
+	// pflags is the full set valid flags for a change propagation call.
+	pflags = ptypes | syscall.MS_REC | syscall.MS_SILENT
+
+	// broflags is the combination of bind and read only
+	broflags = syscall.MS_BIND | syscall.MS_RDONLY
+)
+
+// isremount returns true if either device name or flags identify a remount request, false otherwise.
+func isremount(device string, flags uintptr) bool {
+	switch {
+	case flags&syscall.MS_REMOUNT != 0, device == "", device == "none":
```

Comment:
The problem with only looking at MS_REMOUNT is that typical use cases don't do this.

An example of this can be seen in sharedsubtree_linux:
```
Mount(mountPoint, mountPoint, "none", "bind,rw")
```

These are covered in the test case so removing the "" and "none" case you'll see the following failures:
```
--- FAIL: TestMount (0.22s)
    --- FAIL: TestMount/none-bind,slave (0.00s)
        mounter_linux_test.go:82: no such device
    --- FAIL: TestMount/none-bind,rw,slave (0.00s)
        mounter_linux_test.go:82: no such device
    --- FAIL: TestMount/none-bind,ro,slave (0.00s)
        mounter_linux_test.go:82: no such device
--- FAIL: TestSubtreePrivate (0.02s)
        sharedsubtree_linux_test.go:66: no such device
--- FAIL: TestSubtreeShared (0.00s)
        sharedsubtree_linux_test.go:143: no such device
--- FAIL: TestSubtreeSharedSlave (0.00s)
        sharedsubtree_linux_test.go:222: no such device
--- FAIL: TestSubtreeUnbindable (0.00s)
        sharedsubtree_linux_test.go:303: no such device